### PR TITLE
Added CRUD for departments

### PR DESCRIPTION
**Pull request by OlegZapara**

### CODE REVIEW

File: src/departments/departments.service.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const { projects, ...data } = createDepartmentDto;
+    const createdDepartment = await this.prismaService.department.create({
+      data: {
+        ...data,
+        projects: { createMany: { data: projects ?? [] } },
+      },
+      include: { projects: true },
+    });
+    return createdDepartment;
+  }
```
Comment (yarman1) [inline src/departments/departments.service.ts:20]: I'd rather use inline variables in service files like this:
```ts
  async create(createDepartmentDto: CreateDepartmentDto) {
    const { projects, ...data } = createDepartmentDto;
    return this.prismaService.department.create({
      data: {
        ...data,
        projects: {createMany: {data: projects ?? []}},
      },
      include: {projects: true},
    });
  }
```
The method names in this file are pretty clear, so it shouldn't hurt readability.

**Review by yarman1 — APPROVED** [review]
Overall this is very nice and clean code

File: src/departments/departments.service.ts (hunk 1 above)
Comment (OlegZapara) [reply src/departments/departments.service.ts]: Readability wasn't the case there. In my opinion code with temporary variable like that is easier to debug (you can set breakpoint near variable and see the results). Anyway, I don't think we need it here either

File: src/departments/projects/projects.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+      data: createProjectDto,
+    });
+    return createdProject;
+  }
+
+  async findAll(departmentId: number) {
+    return await this.prismaService.departmentProject.findMany({
+      where: { departmentId },
+    });
+  }
```
[the remark this answers is not in the document]
Comment (OlegZapara) [reply src/departments/projects/projects.service.ts]: Makes sense. Just old memory from C# :)

**Review by yarman1 — APPROVED** [review]

**Pull request merged**